### PR TITLE
task(settings): Disable 2FA and Recovery Code actions when account has no password

### PR DESCRIPTION
**Pull request by dschom**
## Because
- With third party oauth, we might enter into a state where the user has authenticated, but has not supplied a password yet. In this case, setting up recovery codes or 2FA isn’t applicable.

## This pull request
- Introduces hasPassword flag on account model
- Guards routes associated to 2FA and recovery codes in the event the account does not have a password
- Disables 2FA and recovery code action buttons in the event a user’s account has no password set.
- Provides a tooltip hint on the disabled action button that a password must be set in order to use certain features or sync.

## Issue that this pull request solves

Closes: #12774

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

image

## Other Information

To test this state, login with a third party oauth account (apple or google) for the first time. This will result in a state where you have been authenticated, but haven't set a password yet.

### CODE REVIEW

File: packages/fxa-settings/src/components/UnitRowRecoveryKey/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/components/UnitRowRecoveryKey/index.tsx:61]: I changed this in my PR to "Firefox Sync" since that's how we refer to it it user-facing bits. I've also called the ID `security-set-password` - maybe copy that so I don't have to change it?

Alternatively, I think this could be more descriptive with something like, "Set a password before creating a recovery key," and then, "Set a password before adding two-step authentication" for the other?

File: packages/fxa-settings/src/models/Account.ts
Comment (LZoog) [inline packages/fxa-settings/src/models/Account.ts:282]: Should this be `!== null`? But, nice👌, let's get your PR in so I can rebase and use this instead of my `hasPassword = passwordCreated > 0` check in `Security/index.tsx` in my PR.

File: packages/fxa-settings/src/components/UnitRow/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/components/UnitRow/index.tsx:133]: Hmmm, I don't think we want the button itself to read "Disable" or "Disabled", Vesta just mentioned to me that she wanted the buttons (well, links, but I know a link can't be "disabled") "greyed out." I think we want the button to be disabled, but still read "Create" or "Add" so the user can see what the disabled action _is_. I think that's also better for screenreaders, because otherwise I believe (would need to double check on when "title" is read) this button will read "button, disabled, disabled, [title]" while reading "button, Create, disabled, [title]" is going to give those users slightly more context for what they've landed on.

File: packages/fxa-settings/src/models/Account.ts
Comment (dschom) [reply packages/fxa-settings/src/models/Account.ts:282]: I think it should be != which covers both the null case and the undefined case.

File: packages/fxa-settings/src/components/UnitRow/index.tsx
Comment (dschom) [reply packages/fxa-settings/src/components/UnitRow/index.tsx:133]: I can do this. I must say, however, I have some concerns about the mobile experience here. The grayed out button isn't super obvious and the title (ie tooltip) doesn't really do much for mobile users, since the hover event isn't really a thing.

**Review by LZoog — APPROVED** [review]
LGTM! I'll rebase my PR on this once it's merged.

**Pull request merged**
